src/server/server.ts

```diff
@@ -210,6 +212,40 @@ registerRoute('post', '/compile/snippet', compileCommon);
  */
 registerRoute('post', '/compile/page', (req, res) => compileCommon(req, res, true /*wrapWithRunnerChrome*/));
 
+registerRoute('get', '/open-in-playground/:host/:type/:id/:filename', async (req, res) => {
+    let relativePath;
+    switch (req.params.host.toUpperCase()) {
+        case 'EXCEL':
+            relativePath = 'xl';
+            break;
+        case 'WORD':
+            relativePath = 'word';
+            break;
+        case 'POWERPOINT':
+            relativePath = 'ppt';
+            break;
+        default:
+            return;
+    }
+
+    return Promise.resolve(getVersionNumber()).then(versionNumber => {
+        let extractDirName = path.resolve(__dirname, `test${(new Date()).getTime()}`);
+        let zip = fs.createReadStream(path.resolve(__dirname, 'Doc1.zip')).pipe(unzip.Extract({ path: extractDirName }));
+        zip.on('close', () => {
+            let xmlFileName = `${extractDirName}/${relativePath}/webextensions/webextension1.xml`;
+            let xmlStringData = fs.readFileSync(xmlFileName, 'utf-8').toString();
```

Comment:
Lets not use readFileSync here as it is a blocking operation, it could slow down our servers ability to process simultaneous requests.